### PR TITLE
Benchmark refactoring, add backwards benchmarks.

### PR DESCRIPTION
**Pull request by csarofeen**

### CODE REVIEW

File: benchmarks/cpp/nvfuser/batch_norm_backward.cpp [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+      int64_t(benchmark_state.iterations()) *
+      (((3 * input.numel()) * int64_t(dataTypeSize(dtype))) +
+       (run_mean.numel() + run_var.numel() + save_mean.numel() +
+        save_var.numel() + weight.numel()) *
+           int64_t(dataTypeSize(DataType::Float))));
+}
+
+//------------------------------------------------------------------------------
+
+static void Baseline_BatchNorm_BWD(
```
Comment (rdspring1) [inline benchmarks/cpp/nvfuser/batch_norm_backward.cpp:124]: Maybe change the function name to 'Baseline_CuDNN_BatchNorm_BWD' to distinguish it from the native_batch_norm functions.

File: benchmarks/cpp/nvfuser/layer_norm_backward.cpp [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+
+static void NvFuserScheduler_LayerNorm_BWD(
+    benchmark::State& benchmark_state,
+    FusionExecutorCache* fusion_executor_cache,
+    DataType dtype) {
+  TORCH_INTERNAL_ASSERT(dtype == DataType::Float || dtype == DataType::Half);
+
+  std::vector<int64_t> input_shape{
+      benchmark_state.range(0), benchmark_state.range(1)};
+  const float kEps = 1e-5;
```
Comment (rdspring1) [inline benchmarks/cpp/nvfuser/layer_norm_backward.cpp]: `kEps` isn't used here.

**Review by rdspring1 — APPROVED** [review]
LGTM

nitpick: I saw `// ->RangeMultiplier(2)` in several benchmarks. Maybe they should be removed if they aren't being used.

**Comment by csarofeen** [comment]
`// ->RangeMultiplier(2)` is kept in for convenience as when we're doing nightlies doing a multiplier of 8 will save a lot of time but when doing detailed perf work we want it at 2.

File: benchmarks/cpp/nvfuser/batch_norm_backward.cpp (hunk 1 above)
Comment (csarofeen) [reply benchmarks/cpp/nvfuser/batch_norm_backward.cpp:124]: native should be using cuDNN as well. Will mark both as cuDNN.

**Pull request merged**